worlds/ff1/Client.py

```diff
@@ -181,64 +186,76 @@ async def location_check(self, ctx):
                     f'{len(ctx.missing_locations) + len(ctx.checked_locations)})')
                 await ctx.send_msgs([{"cmd": "LocationChecks", "locations": [location]}])
 
-
-
-    async def received_items_check(self, ctx):
+    async def received_items_check(self, ctx: "BizHawkClientContext") -> None:
+        assert self.consumable_stack_amounts, "shouldn't call this function without reading consumable_stack_amounts"
         items_received_count = await self.read_sram_value(ctx, items_obtained)
+        write_list: List[tuple[int, List[int], str]] = []
```

Comment:
this should be fine now that 3.8/3.9 is dropped
```suggestion
        write_list: list[tuple[int, list[int], str]] = []
```